### DISCUSSION TITLE
Wazuh Agent Issues

### DISCUSSION THREAD

**Discussion by B3DTech** [Security-Onion-Solutions/securityonion#4175] [Unsupported Versions]
SO 2.3.50

So I'm not too sure when this broke.  But all my agents are reporting:

WARNING: (4101): Waiting for server reply (not started)

In the Wazuh server logs, I see many, many logs reporting:

Warning: (1408): Invalid ID 196 for the source ip: '10.85.164.220' (name 'unknown').

The ports are open.
# iptables -nL | grep 1514
ACCEPT     tcp  --  0.0.0.0/0            172.17.0.11          tcp dpt:1514
ACCEPT     udp  --  0.0.0.0/0            172.17.0.11          udp dpt:1514
ACCEPT     udp  --  172.23.0.0/21        0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  172.23.0.0/21        0.0.0.0/0            tcp dpt:1514
ACCEPT     udp  --  0.0.0.0/0            0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  0.0.0.0/0            0.0.0.0/0            tcp dpt:1514
ACCEPT     udp  --  10.85.190.50         0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  10.85.190.50         0.0.0.0/0            tcp dpt:1514
ACCEPT     udp  --  127.0.0.1            0.0.0.0/0            udp dpt:1514
ACCEPT     tcp  --  127.0.0.1            0.0.0.0/0            tcp dpt:1514

**Comment by B3DTech** [comment]
So one strange thing I noticed - when I look at the Wazuh Agent Manager UI, all my agent names have a 2 appended to them.

Hostname:  isadmin02
Agent:  isadmin022 (207) - any

Not where why it's appending the 2 to the name, but it's happening on all agents.

**Comment by B3DTech** [comment]
So - I needed to remove the client.keys file from my agent and re-register the agent to SO.

When looking at registered agents, I didn't have any listed, active or disconnected.  That is super strange.  I'm guessing something happened during an upgrade that blew away the agent registration information?

And even though Wazuh documentation says to update the port to 1514 TCP in the config, you still need to use 1514 UDP in the agent.

Once I did both of these and restart the agent, it was communicating successfully.

**Comment by Mastadamus** [comment]
Looks like same thing is happening to me. Did you essentially reregister all agents?

**Reply by Mastadamus** [reply]
I restarted the wazuh service on the Security Onion and all agents reconnected successfully. I was getting the exact same errors on all agents previously and I was receiving no ossec module logs or alerts prior to the restart. It was showing green in so-status but a restart fixed the problem.
